eo-maven-plugin/src/main/java/org/eolang/maven/DiscoverMojo.java

```diff
@@ -94,26 +95,33 @@ public void exec() throws IOException {
     private Collection<String> discover(final Path file)
         throws FileNotFoundException {
         final XML xml = new XMLDocument(file);
+        final List<String> base = new ListOf<>(
+            "//o[",
+            "not(starts-with(@base,'.'))",
+            "and @base != 'Q'",
+            "and @base != '^'",
+            "and @base != '$'",
+            "and @base != '&'",
+            "and not(@ref)"
+        );
+        final List<String> regular = new ListOf<>(base);
+        if (this.withVersions) {
+            regular.add("and not(@ver)");
+        }
+        regular.add("]/@base");
         final Collection<String> names = new TreeSet<>(
```

Comment:
@maxonfjvipon Do we need to use `TreeSet` here? Looks like we can just remove it:
```
final Collection<String> names = DiscoverMojo.names(xml, regular)
```

Reply:
@volodya-lombrozo I believe we can't because we use `/@base` in xpath which is not unique. Correct me if I'm wrong

Reply:
@maxonfjvipon Maybe we can return `Set` directly from `names` method then?
```java
   private static List<String> names(final XML xml, final List<String> xpath) {
         return new SetOf<>(
             new Filtered<>(
                 obj -> !obj.isEmpty(),
                 xml.xpath(
                     String.join(
                         " ",
                         xpath
                     )
                 )
             )
         );
     }
```
Looks like we don't use `List` features, so what is the point to wrap collections to each other?

Reply:
@volodya-lombrozo `TreeSet` is used to save order of strings being added, I don't really know if it's important but `TreeSet` was used in previous implementation so I don't touch it. `LisfOf` is used to decorate `Filtered` which can't be added to `TreeSet` directly.

Reply:
@maxonfjvipon Can we try to use only `SetOf` without `TreeSet`?  I don't think we need order here. If we need - tests will show it.